models/models.js

```diff
@@ -24,3 +24,19 @@ exports.selectReviewById = (review_id) => {
     return review;
   });
 };
+
+exports.selectCommentsByReviewId = (review_id) => {
+  return db
+    .query(`SELECT * FROM comments WHERE review_id = $1 ORDER BY created_at desc`, [review_id])
+    .then(({ rows }) => {
+      if (rows.length === 0) {
+        return this.selectReviewById(review_id);
```

Comment:
you're on the right lines with using the `selectReviewById` model function - however I would recommend calling the function inside your controller `getCommentsByReviewId` 

By using either a `.then` chain or a `Promise.all` you can call more than one model in the controller, if the `selectReviewById` function returns a `Promise.reject` because the `id` is not found - this will go immediately to the `.catch` block of the controller and return the appropriate error